src/content/doc-surrealdb/reference-guide/full-text-search.mdx

```diff
@@ -1,47 +1,532 @@
-An analyzer includes [tokenizers](/docs/surrealql/statements/define/analyzer#tokenizers) and [filters](/docs/surrealql/statements/define/analyzer#filters) which help break down text into manageable tokens and refine the search.
+This will give you a comprehensive overview of the different ways to work with text in SurrealDB and which one to use in your specific use case.
+
+## Comparing and sorting text
+
+### In `SELECT` queries
+
+The following example shows a few records created from an array of strings in an order that is sorted to the human eye: lowest to highest numbers, then A to Z.
+
+```surql
+FOR $word IN ['1', '2', '11', 'Ábaco', 'kitty', 'Zoo'] {
+	CREATE data SET val = $word;
+};
+```
+
+Inside a `SELECT` query, an `ORDER BY` clause can be used to order the output by one or more field names. For the above data, an ordered `SELECT` query looks like this.
+
+```surql
+SELECT VALUE val FROM data ORDER BY val;
+```
+
+However, in the case of strings, sorting is done by Unicode rank which often leads to output that seems out of order to the human eye. The output of the above query shows the following:
+
+```surql title="Output"
+[ '1', '11', '2', 'Zoo', 'kitty', 'Ábaco' ]
+```
+
+This is because:
+
+* '11' is ordered before '2', because the first character in the string '2' is greater than the first character in the string '1'.
+* 'Zoo' is ordered before 'kitty', because the first character in the string 'Zoo' is 'Z', number 0059 in the [list of Unicode characters](https://en.wikipedia.org/wiki/List_of_Unicode_characters#Basic_Latin). A lowercase 'k' is 0076 on the list and thus "greater", while the 'Á', registered as the "Latin Capital letter A with acute", is 0129 on the list.
+
+To sort strings in a more natural manner to the human eye, the keywords [`NUMERIC` and `COLLATE` (or both) can be used](/docs/surrealql/statements/select#sort-records-using-the-order-by-clause). `NUMERIC` will instruct strings that parse into numbers to be treated as such.
+
+```surql
+SELECT VALUE val FROM data ORDER BY val NUMERIC;
+```
+
+```surql title="Numberic strings now sorted as numbers"
+[ '1', '2', '11', 'Zoo', 'kitty', 'Ábaco' ]
+```
+
+`COLLATE` instructs unicode strings to sort by alphabetic order, rather than Unicode order.
+
+```surql
+SELECT VALUE val FROM data ORDER BY val COLLATE;
+```
+
+```surql title="Output"
+[ '1', '11', '2', 'Ábaco', 'kitty', 'Zoo' ]
+```
+
+And for the data in this example, `COLLATE NUMERIC` is likely what will be desired.
+
+```surql
+SELECT VALUE val FROM data ORDER BY val COLLATE NUMERIC;
+```
+
+```surql title="Output"
+[ '1', '2', '11', 'Ábaco', 'kitty', 'Zoo' ]
+```
+
+### Sorting functions
+
+As of SurrealDB `2.2.2`, the functions [`array::sort_natural()`, `array::sort_lexical()`, and `array::sort_lexical_natural()`](/docs/surrealql/functions/database/array) can be used on ad-hoc data to return the same output as the `COLLATE` and `NUMERIC` clauses in a [`SELECT` statement](/docs/surrealql/statements/select).
```

Comment:
@Dhghomon these functions aren't documented yet. Would you mind adding them and updating the link I added here 🙏🏾